### ISSUE TITLE
Always get the same prediction

### ISSUE THREAD

**Issue by staywithme23** [asanakoy/deeppose_tf#2]
Hi after training on LSP dataset.
When I output the prediction for different image, the output joint location are really close.
The reason could be I just train for 20000 iteration.
Could you let me know the reasonable training iteration or could you provide trained weights?
Thx
<img width="364" alt="screen shot 2017-04-05 at 11 40 23 am" src="https://cloud.githubusercontent.com/assets/12218969/24721300/bd7e9ce4-19f4-11e7-9511-c0a7140426f3.png">
<img width="241" alt="screen shot 2017-04-05 at 11 41 28 am" src="https://cloud.githubusercontent.com/assets/12218969/24721324/d7e4aa2e-19f4-11e7-844f-fe11144fffb4.png">

**Comment by asanakoy** [comment]
@adwin5 Hi,
You should train for at least 300k-400k iterations. 20k iteration is not enough.
Check PCP score and compare with the table I provided to be able to see if you converged or not.

**Comment by staywithme23** [comment]
@asanakoy Thank you so much.
Let me try longer training.

**Comment by staywithme23** [comment]
@asanakoy
1, can you refer me where to see PCP score you provided?
2, I am using train_lsp_alexnet_scratch.sh, is that also around 300K?
Thanks in advance

**Comment by asanakoy** [comment]
@adwin5  Ah, sorry, I haven't added them yet :) You should get PCP score something around 55-56% if you train starting with imagenet initialization.
I'm planning to add tables with our results and upload models next week.

**Comment by staywithme23** [comment]
@asanakoy 
Thank you. I appreciate a lot.
Below is an example output. I think you are saying the "val/mPCP" value, right?
Now is 0.027. and I expect it to achieve 55%? or what is the expected PCP score that means converge.
//
Step	 4500	 val/mPCP	 0.027
Step 4500 val/parts_PCP:
Head	Torso	U Arm	L Arm	U Leg	L Leg	mean
0.002	0.132	0.009	0.005	0.012	0.002	0.027
//

**Comment by asanakoy** [comment]
@adwin5 , usually you monitor the loss and it converged when the loss cease decreasing. In case of LSP and Alexnet model loss will decrease at least to the value 0.03 
In my implementation  val/ is  just a subset of train, I did it for sanity check only. ` val/mPCP` should be quit high though (not less than 70%).
To get a test PCP score see output:
`Step N test/mPCP VAL`

**Comment by staywithme23** [comment]
@asanakoy Got it.
I think training from scratch is way too slow than from imagenet.
However, after setting the path to pre-trained alexnet and execute `train_lsp_alexnet_imagenet.sh`
I got the error in below.
Not sure why I can't open `/home/adwin/Project/deeppose_tf/weights/bvlc_alexnet.tf`
//
DataLossError (see above for traceback): Unable to open table file /home/adwin/Project/deeppose_tf/weights/bvlc_alexnet.tf: Data loss: file is too short to be an sstable: perhaps your file is in a different file format and you need to use a different restore operator?
	 [[Node: save/RestoreV2 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save/Const_0, save/RestoreV2/tensor_names, save/RestoreV2/shape_and_slices)]]
	 [[Node: save/RestoreV2_2/_25 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/gpu:0", send_device="/job:localhost/replica:0/task:0/cpu:0", send_device_incarnation=1, tensor_name="edge_110_save/RestoreV2_2", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/gpu:0"]()]]

**Comment by staywithme23** [comment]
@asanakoy 
I debug it. The reason for DataLossError is my `bvlc_alexnet.tf` file is corrupted

**Comment by staywithme23** [comment]
Now training makes sense. Thx.

**Comment by asanakoy** [comment]
@adwin5 You can check tables with the results in the [Readme.md](https://github.com/asanakoy/deeppose_tf/blob/master/README.md)

**Comment by dupsys** [comment]
Hi a

dwin,

Good day, please will you be able to help with regards to the attached code. I got an 

Error when checking model input: expected input_1 to have 2 dimensions, but got array with shape (307551, 180, 68)

I am new in keras and I have tried my best to get help online but I got your email from GitHub. I'm trying to implement Tweet2Vec, where it can be used for feature extraction as the author describe it.

Best Regards

**Comment by dupsys** [comment]
Hi adwin5
[codesegment.txt](https://github.com/asanakoy/deeppose_tf/files/935088/codesegment.txt)

Good day, please will you be able to help with regards to the attached code. I got an 

Error when checking model input: expected input_1 to have 2 dimensions, but got array with shape (307551, 180, 68)

I am new in keras and I have tried my best to get help online but I got your email from GitHub. I'm trying to implement Tweet2Vec, where it can be used for feature extraction as the author describe it.

Best Regards
